src/classfile/parse.rs

```diff
@@ -38,27 +38,42 @@ impl ClassFileParser {
     pub fn bytes(path: String) -> Result<Vec<u8>> {
         info!("opening classfile '{}' to read bytes", path);
 
-        let mut path = path;
-
-        if path.starts_with("java/") {
+        let path = if path.starts_with("java/") {
             debug!("java stdlib detected, altering path");
+            format!("src/stdlib/{}", path)
+        } else {
+            path
+        };
+        let path_with_ext = format!("{}.class", path);
+
+        debug!("loading bytes from '{}'", path_with_ext);
+
+        let mut file_handle = match File::open(&path_with_ext) {
```

Comment:
We should attempt to load from the original input first, as this will almost always be in the correct form. This will save unneeded IO.

```suggestion
        let mut file_handle = match File::open(&path) {
```